src/Instrumentation/Laravel/src/HttpInstrumentation.php

```diff
@@ -79,6 +80,15 @@ public static function register(CachedInstrumentation $instrumentation): void
                     $span->setAttribute(TraceAttributes::NETWORK_PROTOCOL_VERSION, $response->getProtocolVersion());
                     $span->setAttribute(TraceAttributes::HTTP_RESPONSE_BODY_SIZE, $response->headers->get('Content-Length'));
                 }
+                if(($route = Route::getCurrentRoute()?->uri()) !== null) {
```

Comment:
Can this be achieved in the pre hook [here](https://github.com/open-telemetry/opentelemetry-php-contrib/blob/720adadba6e8081a301a857c93ef7f1dc56a81b7/src/Instrumentation/Laravel/src/HttpInstrumentation.php#L32) instead?

The `$request` object is already in scope and should be able to provide the URI there.

Or `$request->route()` for the named route?

Reply:
@LauJosefsen ahh, that might make sense then.

Failing that, you could pull a request object out of the params in the post-hook in the same way as the pre-hook:
```php
$request = ($params[0] instanceof Request) ? $params[0] : null;
```
(anything to avoid the facade usage! 👀 )

Reply:
I can confirm the route is unavailable in the pre hook, unless we call the router manually, which I think is a bad idea.

`$request->route` can be used instead of the facade. The return type is not very nice though, being `Route|null|object|string` . I can do an implementation that sets it to the string returned if string, the route uri if Route, and ignore it if it is a non-Route-object or null?  

Something like this:

```php
if(($route = self::getRouteName($request)) !== null){
    $span->updateName(sprintf('%s %s', $request?->method() ?? 'unknown', $route));
}

private static function getRouteName(Request $request): string {
    $route = $request->route();
    if($route instanceof Route){
        return $route->uri();
    }
    if(is_string($route)){
        return $route;
    }
    return null;
}
```

Reply:
Agreed, let's stick to the post-hook :+1: 

You can end up with something like this and avoid facade usage:

```php
$request = ($params[0] instanceof Request) ? $params[0] : null;
if ($request && $routeName = $request->route()?->uri() ?: $request->route() ?: null) {
    $span->updateName(sprintf('%s %s', $request?->method() ?? 'unknown', '/'. ltrim($routeName, '/')));
}
```

(disclaimer: facade avoidance is a personal preference :sweat_smile: ) 

@LauJosefsen I'm looking forward to seeing this land, thanks for looking into it!